internal/analytics/client/async/client.go

```diff
@@ -91,9 +93,13 @@ func (a *Client) Wait() {
 }
 
 func (a *Client) sendEvent(category, action, label string, dims ...*dimensions.Values) error {
-	if a.closed {
+	switch {
+	case a.closed:
 		logging.Debug("Client is closed, not sending event")
 		return nil
+	case a.cfg.IsSet(constants.ReportAnalyticsConfig) && !a.cfg.GetBool(constants.ReportAnalyticsConfig):
+		logging.Debug("Analytics are disabled; not sending event")
+		return nil
```

Comment:
This should happen in the constructor instead, otherwise we're checking the config on each event as well as opening ourselves up to race conditions with events that send at the end of the process.

See the sync client for example:

http://github.com/ActiveState/cli/blob/5e633be5fc4afb884136a2fd871abc75622e2cd7/internal/analytics/client/sync/client.go#L91-L91

Reply:
Yes, you're right.